android/java/org/chromium/chrome/browser/vpn/utils/BraveVpnProfileUtils.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser.vpn.utils;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.ConnectivityManager;
+import android.net.NetworkCapabilities;
+import android.net.NetworkInfo;
+import android.os.Build;
+
+import androidx.core.content.ContextCompat;
+
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.vpn.wireguard.WireguardService;
+import org.chromium.chrome.browser.vpn.wireguard.WireguardUtils;
+import org.chromium.ui.widget.Toast;
+
+public class BraveVpnProfileUtils {
+    private static BraveVpnProfileUtils sBraveVpnProfileUtils;
+
+    public static BraveVpnProfileUtils getInstance() {
```

Comment:
static methods should always be reentrant, in particular any `getInstance` singleton helper